Dockerfile

```
@@ -23,6 +23,7 @@ FROM node:16-slim AS superset-node
 
 ARG NPM_BUILD_CMD="build"
 ENV BUILD_CMD=${NPM_BUILD_CMD}
+ENV PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true
```

Can we default this to `false`?
```suggestion
ENV PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=false
```

---

Why do we need to add `Chromium` to the image?
What do we use `Puppeteer` for?

---

Puppeteer is a dependency of Apitools - that runs a cron in github actions

Not actually needed in the docker image build - skipping the download in the dockerfile wont affect the apitools tests

---

Ok, I read through some docs. Am I right in the following assumptions?
- The env variable only blocks `puppeteer` from downloading chrome when the docker image is being built during `npm ci`
- We're not depending on this functionality indirectly despite pulling in `puppeteer` in our npm deps

@nisheldo please add a summary of what kind of error you're getting without this fix, preferably a dump of the docker build error you're seeing.

---

All our team also uses Mac M1 and we have problem with ´puppeteer´ lib on arm64 image build and we skip downloading of the lib.
@michael-s-molina @kgabryje Can we speed up downloading of the UI libs while building a docker image? For example: skip all libs for unit test when we run ´npn ci´?

---

@villebro - this is the error you get without the fix


` > [superset-node 7/8] RUN /frontend-mem-nag.sh         && cd /app/superset-frontend         && npm ci:
#18 0.141 Memory check Ok [12.1407 GB free]
#18 108.2 npm notice
#18 108.2 npm notice New major version of npm available! 7.24.2 -> 9.2.0
#18 108.2 npm notice Changelog: <https://github.com/npm/cli/releases/tag/v9.2.0>
#18 108.2 npm notice Run `npm install -g npm@9.2.0` to update!
#18 108.2 npm notice
#18 108.2 npm ERR! code 1
#18 108.2 npm ERR! path /app/superset-frontend/node_modules/puppeteer
#18 108.2 npm ERR! command failed
#18 108.2 npm ERR! command sh -c node install.js
#18 108.2 npm ERR! The chromium binary is not available for arm64.
#18 108.2 npm ERR! If you are on Ubuntu, you can install with:
#18 108.2 npm ERR!
#18 108.2 npm ERR!  sudo apt install chromium
#18 108.2 npm ERR!
#18 108.2 npm ERR!
#18 108.2 npm ERR!  sudo apt install chromium-browser
#18 108.2 npm ERR!
#18 108.2 npm ERR! /app/superset-frontend/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserFetcher.js:115
#18 108.2 npm ERR!                     throw new Error();
#18 108.2 npm ERR!                     ^
#18 108.2 npm ERR!
#18 108.2 npm ERR! Error
#18 108.2 npm ERR!     at /app/superset-frontend/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserFetcher.js:115:27
#18 108.2 npm ERR!     at FSReqCallback.oncomplete (node:fs:202:21)
#18 108.3
#18 108.3 npm ERR! A complete log of this run can be found in:
#18 108.3 npm ERR!     /root/.npm/_logs/2023-01-11T17_24_26_860Z-debug.log`

---

Thanks @nisheldo - can you add it to the description to improve discoverability?

---

Funny thing is, I'm on an M1, too, but haven't run into this, so I guess I'm not using the `Dockerfile` that much locally 😁

---

@villebro  - done

---

@villebro Can we merge it?

---

@rusackas I think we can merge it